### PR TITLE
DEV: Improves Error msg when constructing a Timedelta

### LINKED ISSUE #54059: DEV: Error in `main` branch when subtracting a valid numpy timedelta from a Series of timestamps
```python
import pandas as pd
import numpy as np
s1 = pd.Series(data=pd.date_range("1/1/2020", "2/1/2020"))
td = np.timedelta64(1, "M")
s1-td
```
With version 2.1.0.dev0+1159.gc126eeb392 and numpy 1.24.3, get the following error:
```text
  File "C:\Code\pandas_dev\pandas\pandas\core\ops\common.py", line 85, in new_method
    return method(self, other)
  File "C:\Code\pandas_dev\pandas\pandas\core\arraylike.py", line 194, in __sub__
    return self._arith_method(other, operator.sub)
  File "C:\Code\pandas_dev\pandas\pandas\core\series.py", line 5735, in _arith_method
    return base.IndexOpsMixin._arith_method(self, other, op)
  File "C:\Code\pandas_dev\pandas\pandas\core\base.py", line 1342, in _arith_method
    rvalues = ops.maybe_prepare_scalar_for_op(rvalues, lvalues.shape)
  File "C:\Code\pandas_dev\pandas\pandas\core\ops\array_ops.py", line 575, in maybe_prepare_scalar_for_op
    return Timedelta(obj)
  File "timedeltas.pyx", line 1846, in pandas._libs.tslibs.timedeltas.Timedelta.__new__
ValueError:  cannot construct a Timedelta from a unit M
```

Tagging @mcgeestocks  since last commit related to this code made by him and @mroeschke since he approved the PR #53421

### PR DESCRIPTION
**Pull request by mcgeestocks**
- [x] closes #54059 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -1843,7 +1843,10 @@ class Timedelta(_Timedelta):
                              NPY_DATETIMEUNIT.NPY_FR_W,
                              NPY_DATETIMEUNIT.NPY_FR_GENERIC]):
                 err = npy_unit_to_abbrev(reso)
-                raise ValueError(f" cannot construct a Timedelta from a unit {err}")
+                raise ValueError(
+                    "Units 'M', 'Y', and 'y' are no longer supported, as they do not "
+                    "represent unambiguous timedelta values durations."
+                    "Allowed unites are 'W', 'D', 'h', 'm', 's', 'ms', 'us', 'ns'")
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timedeltas.pyx:1849]: "unites" 😉

**Review by MarcoGorelli — COMMENTED** [review]
Nice! Just need a whatsnew note (2.1.0) and this can go in

I think there's an "other" section where similar error message improvements are sometimes reported

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 2]
```diff
@@ -1843,7 +1843,10 @@ class Timedelta(_Timedelta):
                              NPY_DATETIMEUNIT.NPY_FR_W,
                              NPY_DATETIMEUNIT.NPY_FR_GENERIC]):
                 err = npy_unit_to_abbrev(reso)
-                raise ValueError(f" cannot construct a Timedelta from a unit {err}")
+                raise ValueError(
+                    "Units 'M', 'Y', and 'y' are no longer supported, as they do not "
```
Comment (mroeschke) [inline pandas/_libs/tslibs/timedeltas.pyx:1847]: A message like `f"Unit {err} is not supported. Allowed units are ..."` might be more general. A user here might pass a np.timedelta object that is higher resolution than ns for example

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/timedeltas.pyx:1847]: that's a good point, thanks, numpy allows picoseconds and other very small things

Comment (mcgeestocks) [reply pandas/_libs/tslibs/timedeltas.pyx]: Updated with suggestions but still kept the reference to ambiguous time periods being no longer supported.

File: doc/source/whatsnew/v2.1.0.rst [hunk 3]
```diff
@@ -111,6 +111,7 @@ Other enhancements
 - Improve error message when setting :class:`DataFrame` with wrong number of columns through :meth:`DataFrame.isetitem` (:issue:`51701`)
 - Improved error handling when using :meth:`DataFrame.to_json` with incompatible ``index`` and ``orient`` arguments (:issue:`52143`)
 - Improved error message when creating a DataFrame with empty data (0 rows), no index and an incorrect number of columns. (:issue:`52084`)
+- Improved error message when creating a Timedelta from a ambiguous time period (:issue:`54059`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:114]: We don't need a whatsnew entry for this change since the original change was going to be new in 2.1

**Comment by mcgeestocks** [comment]
Looks good to go @mroeschke / @MarcoGorelli

**Review by mroeschke — APPROVED** [review]

**Pull request merged**